File: jmeter-test/blocking-test/src/main/java/study/spring/jmeter/blocking/blockingtest/TestController.java
package study.spring.jmeter.blocking.blockingtest;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class TestController {

  @GetMapping("/test")
  public String test() throws InterruptedException {
    Thread.sleep(500);
    return "test";
  }
}

File: async-thread-pool/src/main/kotlin/study/spring/asyncthreadpool/presentation/TestController.kt
package study.spring.asyncthreadpool.presentation

import org.springframework.core.task.SimpleAsyncTaskExecutor
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController
import study.spring.asyncthreadpool.service.TestService

var index = 1

@RestController
class TestController(val testService: TestService) {

    @PostMapping("/test/sleep/{sleepSecond}")
    fun testWithSleep(@PathVariable sleepSecond: Int): ResponseEntity<String> {
        val currentIndex = index
        index++
        val threadName = Thread.currentThread().name
        println("[메인-시작-${currentIndex}] ${threadName}")
        testService.testWithSleep(sleepSecond, currentIndex)
        return ResponseEntity.ok().build()
    }
}